src/cli/main.py

```diff
@@ -197,6 +199,60 @@ def get_job(ctx: click.Context, name: str) -> None:
     _output_result(resp)
 
 
+def make_job_create_request(
+    marketplace_image_urn: str,
+    vhd_sas_url: str,
+    job_name: str,
+    template_name: str,
+    resource_group: str,
+    subscription_id: str,
+    vm_size: str,
+    test_priorities: list[str],
+    test_cases: list[str],
+    location: str,
+    region: list[str],
+    concurrency: int,
+    vm_generation: int,
+    architecture: str,
+) -> tuple[dict[str, Any], str]:
+    endpoint = _get_endpoint(f"jobs/{job_name}", resource_group, subscription_id)
+
+    image = {
+        "vhdGeneration": vm_generation,
+        "architecture": architecture,
+    }
+    if marketplace_image_urn:
+        image["type"] = "marketplace"
+        image_parts = marketplace_image_urn.split(":")
+        if len(image_parts) != 4:
+            raise ValueError(
+                "Marketplace_image_urn should be in the format of 'publisher:offer:sku:version', "
+                f"got {marketplace_image_urn}"
+            )
+        image["publisher"] = image_parts[0]
+        image["offer"] = image_parts[1]
+        image["sku"] = image_parts[2]
+        image["version"] = image_parts[3]
```

Comment:
Not mandatory, but you could replace this with:

```
image["publisher"], image["offer"], image["sku"], image["version"] = image_parts
```

Reply:
Done